### DISCUSSION TITLE
Cloudflare configuration for pass real ip

### DISCUSSION THREAD

**Discussion by MaksimKiselev** [nginx-proxy/nginx-proxy#2540] [Show and tell]
Hi guys, I didn't find solution for this case and wanna show how I was resolved this.

I have configuration:
Cloudflare [HTTP] => nginx-proxy [HTTP] => site [HTTP]

Troubles:
On site I see IP of nginx-proxy container.

Solution:
Create additional configuration file like this:
```
set_real_ip_from 103.21.244.0/22;
set_real_ip_from 103.22.200.0/22;
set_real_ip_from 103.31.4.0/22;
set_real_ip_from 104.16.0.0/12;
set_real_ip_from 108.162.192.0/18;
set_real_ip_from 131.0.72.0/22;
set_real_ip_from 141.101.64.0/18;
set_real_ip_from 162.158.0.0/15;
set_real_ip_from 172.64.0.0/13;
set_real_ip_from 173.245.48.0/20;
set_real_ip_from 188.114.96.0/20;
set_real_ip_from 190.93.240.0/20;
set_real_ip_from 197.234.240.0/22;
set_real_ip_from 198.41.128.0/17;
set_real_ip_from 199.27.128.0/21;
set_real_ip_from 2400:cb00::/32;
set_real_ip_from 2606:4700::/32;
set_real_ip_from 2803:f800::/32;
set_real_ip_from 2405:b500::/32;
set_real_ip_from 2405:8100::/32;
real_ip_header CF-Connecting-IP;
```
Actual IP ranges you can find on page https://www.cloudflare.com/ips/

save as `cloudflare.conf` and add volume to nginx-proxy container:
`- /path/to/cloudflare.conf:/etc/nginx/conf.d/my.conf`

Then for site container add to nginx configuration next lines:
```
set_real_ip_from 172.0.0.0/12;
real_ip_header X-Forwarded-For;
```

Ok, it's worked)

Have ideas or recomendations?

**Comment by kamermans** [comment]
Seems reasonable to me, and I can't think of a cleaner way to do it unfortunately.

**Comment by GreenLondon** [comment]
Hello @MaksimKiselev,

Can you be more specific please when it comes to? `- /path/to/cloudflare.conf:/etc/nginx/conf.d/my.conf`

What is the whole command to be used?

Thanks

**Comment by wirespecter** [comment]
@MaksimKiselev This method is not enough. As stated in https://support.cloudflare.com/hc/en-us/articles/200170786-Restoring-original-visitor-IPs the list of cloudflare IPs in `cloudflare.conf` needs to be regularly updated.

**Comment by GramThanos** [comment]
Following @MaksimKiselev comment, using the Cloudflare API I wrote a bash/python3 command to generate the configuration file so that it can be dynamically created and updated. The following code assumes that the `cloudflare.conf` file is located at the current working directory:

```bash
python3 -c "print('\\n'); import requests; response = requests.get('https://api.cloudflare.com/client/v4/ips'); response.raise_for_status(); data = response.json(); x = [print('set_real_ip_from ' + cidr + ';') for cidr in data['result']['ipv4_cidrs']]; x = [print('set_real_ip_from ' + cidr + ';') for cidr in data['result']['ipv6_cidrs']]; print('real_ip_header CF-Connecting-IP;\\n')" > ./cloudflare.conf
```
